src/action_cc.cc

```diff
@@ -966,3 +987,91 @@ NM_ACTION_(nickel_bluetooth) {
             break;
     }
 }
+
+NM_ACTION_(nickel_screenshot) {
+    enum SCREENSHOT_ACTION {
+        CAPTURE = 0b00001,
+    };
+
+    int action = 0;
+    if (!strcmp(arg, "capture")) action |= CAPTURE;
+    else
+        NM_ERR_RET(nullptr, "unknown nickel_screenshot action '%s'", arg);
+
+    QObject* app = nullptr;
+    QKeyEvent* press = nullptr;
+    QKeyEvent* release = nullptr;
+
+    #define vtable_ptr(x) *reinterpret_cast<void**&>(x)
+    #define vtable_target(x) reinterpret_cast<void*>(reinterpret_cast<uintptr_t>(x)+8)
+
+    // Prepare Settings symbols
+    SettingsSymbols settings_syms = prepare_Settings_symbols();
+    auto Settings_Settings = settings_syms.Settings_Settings;
+    auto Settings_SettingsLegacy = settings_syms.Settings_SettingsLegacy;
+    auto Settings_SettingsD = settings_syms.Settings_SettingsD;
+    auto Settings_getSetting = settings_syms.Settings_getSetting;
+    auto Settings_saveSetting = settings_syms.Settings_saveSetting;
+
+    Device *dev = get_Device_getCurrentDevice();
+    Settings *settings = alloca(128); // way larger than it is, but better to be safe
+    if (Settings_Settings)
+        Settings_Settings(settings, dev, false);
+    else if (Settings_SettingsLegacy)
+        Settings_SettingsLegacy(settings, dev);
+
+    QVariant v1;
+
+    //libnickel 4.6 * _ZTV15FeatureSettings
+    void *FeatureSettings_vtable = dlsym(RTLD_DEFAULT, "_ZTV15FeatureSettings");
+    NM_CHECK(nullptr, FeatureSettings_vtable, "could not dlsym the vtable for FeatureSettings");
+    vtable_ptr(settings) = vtable_target(FeatureSettings_vtable);
+
+    QString screenshot_key = QStringLiteral("Screenshots");
+    bool org_screenshot_enabled = false;
+
+    switch (action) {
+        case CAPTURE:
+            // Pressing the Power button on Kobo is the same as pressing the Escape key on keyboard
+            // So to take screenshot, we just need to:
+            // 1. Enable the "Screenshots" feature
+            // 2. Send an "Escape" KeyEvent to the QApplication instance
+
+            // Get Nickel::Application->notify()
+            int (*AppNotify)(void* app, void* receiver, void* event);
+            NM_ACT_XSYM(AppNotify, "_ZN18Nickel3Application6notifyEP7QObjectP6QEvent", "could not dlsym Nickel::Application->notify()");
```

Comment:
Add a symbol test comment (`//libnickel`...) to this with the minimum libnickel version with the symbol.